docs/customizing-install.md

```diff
@@ -8,10 +8,77 @@ nav_order: 5
 1. TOC
 {:toc}
 
+## Creating customized ISO and PXE images
+
+The [`iso customize`](cmd/iso.md#coreos-installer-iso-customize) and
+[`pxe customize`](cmd/pxe.md#coreos-installer-pxe-customize) commands can be
+used to create customized ISO and PXE images with site-specific
+configuration, including the ability to perform unattended installations.
+This is the recommended method for automatically running coreos-installer at
+boot.
+
+For example:
+
+```bash
+# Create customized.iso which:
+# - Automatically installs to /dev/sda
+# - Provisions the installed system with config.ign
+# - Uses network configuration from static-ip.nmconnection
+# - Trusts HTTPS certificates signed by ca.pem
+# - Runs post.sh after installing
+coreos-installer iso customize \
+    --dest-device /dev/sda \
+    --dest-ignition config.ign \
+    --network-keyfile static-ip.nmconnection \
+    --ignition-ca ca.pem \
+    --post-install post.sh \
+    -o custom.iso input.iso
+# Same, but create a customized PXE initramfs image
+coreos-installer pxe customize \
+    --dest-device /dev/sda \
+    --dest-ignition config.ign \
+    --network-keyfile static-ip.nmconnection \
+    --ignition-ca ca.pem \
+    --post-install post.sh \
+    -o custom-initramfs.img input-initramfs.img
+```
+
+Available customizations include:
+
+- Specifying an Ignition config to be applied to the installed system
+  (`--dest-ignition`) or to the live environment (`--live-ignition`).
+- Specifying the device to which the operating system will be installed
+  (`--dest-device`).  If an ISO or PXE image has been customized with
+  `--dest-device`, booting that image will automatically install to the
+  specified disk and reboot the system.
+- Specifying network configuration for both the installed system and the
+  live environment via
+  [NetworkManager keyfiles](https://developer.gnome.org/NetworkManager/stable/nm-settings-keyfile.html)
+  (`--network-keyfile`).
+- Specifying HTTPS certificate authorities to be trusted by Ignition, in
+  both the installed system and the live environment (`--ignition-ca`).
+- Modifying kernel arguments of the installed system (`--dest-karg-append`,
+  `--dest-karg-delete`) or the live ISO environment (`--live-karg-append`,
+  `--live-karg-replace`, `--live-karg-delete`).  There are no `--live-karg`
+  options for the PXE image; modify the PXE boot configuration instead.
+- Running scripts before or after installation (`--pre-install`,
+  `--post-install`).  For example, a pre-install script might run a
+  container that performs hardware validation, or a post-install script
+  might use IPMI to configure the machine to boot from the local disk
+  instead of the network.  Pre-install scripts can change the options
+  processed by coreos-installer, including the choice of destination device,
+  by writing an installer config file to `/etc/coreos/installer.d` (see
+  below).
```

Comment:
In a follow-up, it'd probably be worth fleshing this out with some examples. E.g. a script which uses the MAC address to auto-generate an NM keyfile with static IP to copy in? Or automatically select the destination device based on disk model?

Reply:
https://github.com/coreos/coreos-installer/issues/784